components/script/dom/readablestreambyobreader.rs

```diff
@@ -194,9 +258,130 @@ impl ReadableStreamBYOBReaderMethods<crate::DomTypeHolder> for ReadableStreamBYO
     }
 
     /// <https://streams.spec.whatwg.org/#byob-reader-read>
-    fn Read(&self, _view: CustomAutoRooterGuard<ArrayBufferView>, can_gc: CanGc) -> Rc<Promise> {
-        // TODO
-        Promise::new(&self.reflector_.global(), can_gc)
+    #[allow(unsafe_code)]
+    fn Read(
+        &self,
+        view: CustomAutoRooterGuard<ArrayBufferView>,
+        options: &ReadableStreamBYOBReaderReadOptions,
+        can_gc: CanGc,
+    ) -> Rc<Promise> {
+        let view = HeapBufferSource::<ArrayBufferViewU8>::new(BufferSource::ArrayBufferView(
+            Heap::boxed(unsafe { *view.underlying_object() }),
+        ));
+
+        let cx = GlobalScope::get_cx();
+        // If view.[[ByteLength]] is 0, return a promise rejected with a TypeError exception.
+        if view.byte_length().is_zero() {
+            rooted!(in(*cx) let mut error = UndefinedValue());
+            unsafe {
+                Error::Type("view byte length is 0".to_owned()).to_jsval(
+                    *cx,
+                    &self.global(),
+                    error.handle_mut(),
+                )
+            };
+            return Promise::new_rejected(&self.global(), cx, error.handle()).unwrap();
+        }
+        // If view.[[ViewedArrayBuffer]].[[ArrayBufferByteLength]] is 0,
+        // return a promise rejected with a TypeError exception.
+        if view.viewed_buffer_array_byte_length(cx).is_zero() {
+            rooted!(in(*cx) let mut error = UndefinedValue());
+            unsafe {
+                Error::Type("viewed buffer byte length is 0".to_owned()).to_jsval(
+                    *cx,
+                    &self.global(),
+                    error.handle_mut(),
+                )
+            };
+            return Promise::new_rejected(&self.global(), cx, error.handle()).unwrap();
```

Comment:
Should we perhaps use `reject_error` as discussed in https://github.com/servo/servo/issues/35120#issuecomment-2609510436?